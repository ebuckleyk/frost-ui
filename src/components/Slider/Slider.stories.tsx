import * as React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { Slider } from './Slider';

type ComponentType = React.ComponentProps<typeof Slider>;
const meta: Meta<ComponentType> = {
  component: Slider,
};

export default meta;

type Story = StoryObj<ComponentType>;
export const Demo: Story = {};
