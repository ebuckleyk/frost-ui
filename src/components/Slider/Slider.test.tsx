import * as React from 'react';
import { render } from '@testing-library/react';

import { Slider } from './Slider';

function Component() {
  return (
    <div className="w-[200px]">
      <Slider defaultValue={[50]} max={100} step={1} className={'w-[100%]'} />
    </div>
  );
}

describe('Slider', () => {
  it('should render Slider component to match snapshot', () => {
    const result = render(<Component />);
    expect(result).toMatchSnapshot();
  });
});
