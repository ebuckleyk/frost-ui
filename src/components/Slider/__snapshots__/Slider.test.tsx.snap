// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Slider > should render Slider component to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-[200px]"
      >
        <span
          aria-disabled="false"
          class="relative flex touch-none select-none items-center w-[100%]"
          data-orientation="horizontal"
          dir="ltr"
          style="--radix-slider-thumb-transform: translateX(-50%);"
        >
          <span
            class="relative h-1.5 w-full grow overflow-hidden rounded-full bg-primary/20"
            data-orientation="horizontal"
          >
            <span
              class="absolute h-full bg-primary"
              data-orientation="horizontal"
              style="left: 0%; right: 50%;"
            />
          </span>
          <span
            style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(50% + 0px);"
          >
            <span
              aria-orientation="horizontal"
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="50"
              class="block h-4 w-4 rounded-full border border-primary/50 bg-background shadow transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50"
              data-orientation="horizontal"
              data-radix-collection-item=""
              role="slider"
              style=""
              tabindex="0"
            />
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-[200px]"
    >
      <span
        aria-disabled="false"
        class="relative flex touch-none select-none items-center w-[100%]"
        data-orientation="horizontal"
        dir="ltr"
        style="--radix-slider-thumb-transform: translateX(-50%);"
      >
        <span
          class="relative h-1.5 w-full grow overflow-hidden rounded-full bg-primary/20"
          data-orientation="horizontal"
        >
          <span
            class="absolute h-full bg-primary"
            data-orientation="horizontal"
            style="left: 0%; right: 50%;"
          />
        </span>
        <span
          style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(50% + 0px);"
        >
          <span
            aria-orientation="horizontal"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="50"
            class="block h-4 w-4 rounded-full border border-primary/50 bg-background shadow transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50"
            data-orientation="horizontal"
            data-radix-collection-item=""
            role="slider"
            style=""
            tabindex="0"
          />
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
